core/math/vector2.h

```diff
@@ -251,6 +251,17 @@ _FORCE_INLINE_ bool Vector2::operator!=(const Vector2 &p_vec2) const {
 	return x != p_vec2.x || y != p_vec2.y;
 }
 
+real_t Vector2::normalize() {
+	real_t l = x * x + y * y;
+	if (l != 0) {
+		l = Math::sqrt(l);
+		x /= l;
+		y /= l;
+		return l;
+	}
+	return 0.0;
```

Comment:
Could be simplified:

```suggestion
	real_t l = x * x + y * y;
	if (l != 0) {
		l = Math::sqrt(l);
		x /= l;
		y /= l;
	}
	return l;
```

Reply:
Yup sure. :+1: 
I was intrigued to find in godbolt that the assembly is slightly different.

Even adjusting for the literal issue I mentioned in #56396 (and is probably addressed in a separate PR if we get some agreement on it), I wonder if it is trying to account for signed zero. We'd have to time them to see which was "better", I don't expect there's a lot in it! :grin:

Reply:
`-0.0` would be the only case that is different, so that has to be it. I also checked NaN but that should enter the `if` and then explode things there so it wouldn't be effected by this.